impl/src/lib.rs

```diff
@@ -0,0 +1,241 @@
+extern crate proc_macro;
+extern crate proc_macro2;
+extern crate regex;
+#[macro_use]
+extern crate lazy_static;
+#[macro_use]
+extern crate quote;
+use proc_macro2::TokenStream;
+use proc_macro_hack::proc_macro_hack;
+use regex::Regex;
+
+fn consume_expr(s: &str) -> (&str, String) {
```

Comment:
Sorry about the confusing diff -- everything from here down is moved unmodified from src/lib.rs, with the exception of three occurrences of `#[proc_macro]` which are replaced by `#[proc_macro_hack]`.